app/assets/javascripts/run_claim.coffee

```diff
@@ -0,0 +1,51 @@
+# Format of possible localStorage values is:
+#
+# Standard Claim Token -- a run uploaded while not logged in
+# "claim_tokens/<run_id>": "<claim_token>"
+# e.g. "claim_tokens/gcb": "wYJm4S9uAAra6TMyzLkvhC5y"
+#
+# Dismissed Claim Token -- a Standard Claim Token whose prompt was dismised in the UI 
+# "dismissed_claim_tokens/<run_id>": "<claim_token>"
+# e.g. "dismissed_claim_tokens/gcb": "wYJm4S9uAAra6TMyzLkvhC5y"
+
+activateClaimLink = (claimToken) ->
+  if gon.user != null
+    claimLink = document.getElementById('claim-nav-link')
+    claimLink.href = window.location.pathname + '?claim_token=' + claimToken
+
+  claimLinkContainer = document.getElementById('claim-nav-link-container')
+  claimLinkContainer.style.display = 'inline'
+
+activateClaimPrompt = (claimToken) ->
+  if gon.user != null
+    claimPromptButton = document.getElementById('claim-prompt-button')
+    claimPromptButton.href = window.location.pathname + '?claim_token=' + claimToken
+
+  claimPrompt = document.getElementById('claim-prompt')
+  claimPrompt.style.display = 'block'
+
+$ ->
+  claimTokenKey = 'claim_tokens' + window.location.pathname
```

Comment:
A check for gon.run being defined solves the errors I found.  L34 will error on pages where it isn't defined, and subsequently L39 will error on pages that don't have the 'dismiss-claim-prompt' id present on the page.